app/scripts/metamask-controller.js

```diff
@@ -3964,7 +3964,12 @@ export default class MetamaskController extends EventEmitter {
         // keyring's iframe and have the setting initialized properly
         // Optimistically called to not block MetaMask login due to
         // Ledger Keyring GitHub downtime
-        this.setLedgerTransportPreference();
+        if (this.isKeyringAvailable(LedgerKeyring.type)) {
+          await this.withKeyringForDevice(
+            { name: HardwareDeviceNames.ledger },
+            async (keyring) => this.setLedgerTransportPreference(keyring),
+          );
+        }
```

Comment:
I think this could also be completely removed, since there shouldn't be any Ledger keyring right after creating (or restoring) the vault.

The `this.isKeyringAvailable(LedgerKeyring.type)` condition helps not creating the keyring in case it was not present in the first place

Reply:
Unfortunately we creates a new Ledger keyring for each user irrespective of whether they are Ledger users. This does seem bad, but I am unsure what the consequences would be for stopping this practice. The comment does suggest that setting this preference as soon as possible was critical, but I don't recall why.

Could we consider continuing the questionable practice of creating a Ledger keyring for all users (as we currently do), and stop doing this in a separate PR? So that we can reduce the changes of complications from this refactor.

Reply:
We might also want to drop the `await` here, given the comment above

Reply:
Good point, we can address this in a separate PR. I removed `isKeyringAvailable` from the two occurrences of this practice for Ledger (and the `isKeyringAvailable` function itself, since it is now unused).

Perhaps a good place to set the transport preference would be in `withKeyringForDevice` since it is used for all hardware keyring interactions